docs/madr/decisions/011-multizone-envoy-admin.md

```diff
@@ -0,0 +1,120 @@
+# Multizone Envoy Admin Operations
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/5080
+
+## Context and Problem Statement
+
+Some time ago we introduced functionality that lets user execute Config Dump / Stats / Clusters from the CP.
+This is also available through Global CP in multi zone deployment.
+The problem is that Zone CP from given zone connects to only one instance of Global CP and admin operations are piped through KDS.
+In HA deployment, when you execute the Admin Operation request you don't know which instance you will use.
+
+## Considered Options
+
+* Discovery
+  * Config
+* Communication
+  * API Server
+  * Inter-CP Server
+
+## Pros and Cons of the Options
+
+We need to pass the request to a proper instance, for this we need to consider how we want to communicate between CP instances and how they will discover each other.
+
+## Discovery
+
+### Config
+
+We could gather all instances and IPs in one Config object (ConfigMap on Kubernetes).
+```yaml
+type: Config
```

Comment:
Is it possible to update the Config without conflicts? For example, using `patch` to append an instance at the end of the `instances` list

Reply:
I'm not sure that patch fill works on the content on the ConfigMap. Honestly, I don't think it matters. Again, I don't think this will happen very often and we can just have retries if it fails due to resource conflict.

Reply:
Yeah optimistic concurrency control seems to be the right approach here.
Worst case if a conflict happens k8s will tell you off and you can recompute/retry this is meant to be rare enough.

Can we do something similar in Universal?

Reply:
Yes, we have optimistic locking on Universal. The logic will be the same.